includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1241,6 +1241,11 @@ public function process_payment( $order_id ) {
 				// Set the order status to "pending payment".
 				$order->update_status( 'pending' );
 
+				// phpcs:ignore WordPress.Security.NonceVerification.Missing
+				if ( isset( $_POST['order_has_subscriptions'] ) ) {
+					$order->update_meta_data( 'order_has_subscriptions', '1' );
+				}
```

Comment:
During the preflight check the draft order has no subscription metadata, this is the only approach I found to identify later if the order is a subscription.

Reply:
I was able to remove it on https://github.com/Automattic/woocommerce-payments/pull/9778/commits/27990a4899f5df84a9c5d50b1809edc6e55efda0 .